src/controllers/currency-conversion.controller.js

```diff
@@ -0,0 +1,108 @@
+import acceptedCurrencies from '../data/accepted-currencies.json';
+import log from '../services/logger.service';
+import { getExchangeRate } from '../services/quotation-of-the-day.service';
+
+const calculateExchange = (from, to, amount) => {
+    if (from <= 0 || to <= 0 || amount <= 0) {
+        log(3, 'currency-conversion.controller.js', 'calculateExchange()');
+        return {
+            error: true,
+            convertedValue: '0',
+        };
+    }
+
+    const amountUSD = (amount / from);
```

Comment:
Escolha interessante de *variable-naming*, a idéia não é que o serviço suporte qualquer moeda?

Reply:
@m4n3z40 
Sim, chamar essa variável de `amountUSD` foi uma decisão baseado no fato que `Dollar (UDS)` é nossa moeda lastro. Sendo assim o algoritmo de conversão é sempre baseado nele (o USD). Por esse motivo essa função `calculateExchange` primeiro realizar a conversão do `from -> dollar` e depois conversão do `dollar -> to`, realizando a conversão de *qualquer* `from` para *qualquer* `to`, tendo sempre o USD como *moeda de lastro*.